NumLib/ODESolver/ODETypes.h

```diff
@@ -0,0 +1,212 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef NUMLIB_ODETYPES_H
+#define NUMLIB_ODETYPES_H
+
+// TODO: Move that file somewhere else.
+//       Actually it would be better to put all of this into the MathLib namespace
+//       in the end, because all matrices, vectors and linear solvers reside there.
+//       That will be done during some subsequent cleanup.
+
+#include <initializer_list>
+#include <cassert>
+
+
+// Dense Eigen matrix/vector //////////////////////////////////////////
+// always enabled
+
+#include <Eigen/LU>
+
+namespace NumLib
+{
+
+inline void setMatrix(Eigen::MatrixXd& m,
+                      Eigen::MatrixXd::Index const rows, Eigen::MatrixXd::Index const cols,
+                      std::initializer_list<double> values)
+{
+    using IndexType = Eigen::MatrixXd::Index;
```

Comment:
I don't know what you mean. ^^

> This index type does not correspond to the MathLib::EigenMatrix index type.

That's true, but that's fine, because this method is dealing with `Eigen::MatrixXd`, so the index type must be consistent with that.

> I think the MatrixTraits shall be defined only for MathLib::XXXMatrix/Vector types.

That's not true. (1) `MatrixTraits` are not used in this function. (2) `MatrixTraits` are used in other occasions in the ODESolver lib, thus they must provide the right index type for any matrix type used. Admittedly, currently the `MatrixTraits` are only used in `ODESystem.h`, but the use might be extended, e.g., one could replace many return types of `std::size_t` by `MatrixTraits<...>::Index` in a future PR.